akka/src/main/scala/auth/handlers/AuthActionsHandler.scala

```diff
@@ -56,93 +59,46 @@ class AuthActionsHandler(service: AuthService, emailPasswordServices: EmailPassw
     pathPrefix("actions") {
       switchRoute ~ post {
         (path("changePassword") & userRequired & entity(as[PasswordChange])) { (status, pc) ⇒
-          onSuccess(passwordChangeService.changePassword(status.userId, pc.oldPass.getOrElse(""), pc.newPass)){ _ ⇒
+          onSuccess(passwordChangeService.changePassword(status.userId, pc.oldPass.getOrElse(""), pc.newPass)) { _ ⇒
             complete(status)
           }
-        } ~ path ("startPasswordRecovery") {
-          /*
-          Action.async(authBodyJson[StartPasswordRecover]) {
-    implicit request ⇒
-      for {
-        _ ← passwordRecoveryService.startRecovery(request.body.email)
-      } yield NoContent
-  }
-           */
-          complete("???")
-        } ~ path ("checkPasswordRecovery") {
-          /*
-           Action.apply(authToken[CheckPasswordRecoverToken, PasswordRecoverCommand]) {
-    implicit request ⇒
-      NoContent
-  }
-           */
-          complete("???")
-        } ~ path ("recoverPassword") {
-          /*
-          Action.async(authTokenExpirableJson[FinishPasswordRecover, PasswordRecoverCommand]()) {
-    implicit request ⇒
-      (for {
-        pid ← passwordRecoveryService.finishRecovery(request.body.token.email, request.body.json.newPass)
-        aStatus ← authStatus(pid, false)
-      } yield aStatus.ok) recoverWith {
-        case e: JsonApiFailure ⇒ e.copy(status = 401).resultF
-      }
-  }
-           */
-          complete("???")
-        } ~ path ("verifyEmail") {
-          /*
-          Action.async(authToken[EmailVerifyToken, EmailVerifyCommand]) {
-    implicit request ⇒
-      (for {
-        result ← emailVerifierService.verify(request.body.email)
-      } yield NoContent) recover {
-        case e: JsonApiFailure ⇒ e.result
-        case u: Exception      ⇒ JsonApiFailure(400, "cannot_verify_email", u.getMessage, "auth").result
-      }
-  }
-           */
-          complete("???")
-        } ~ (path ("requestEmailVerify") & userRequired) { s ⇒
+        } ~ (path("startPasswordRecovery") & entity(as[StartPasswordRecover])) { spr =>
+          onSuccess(passwordRecoveryService.startRecovery(spr.email)) {
+            complete(emptyAnswer)
+          }
+        } ~ (path("checkPasswordRecovery") & authTokenCommand[PasswordRecoverCommand, CheckPasswordRecoverToken]) { _ =>
+          complete(emptyAnswer)
+        } ~ (path("recoverPassword") & authTokenExpirableCommand[PasswordRecoverCommand, FinishPasswordRecover]() & userAware) { (cmd, status) =>
+          onSuccess(passwordRecoveryService.finishRecovery(cmd._1.email, cmd._2.newPass)) { pid =>
+            val auth = status.getOrElse(AuthStatus(pid, Seq.empty, None))
```

Comment:
why getOrElse? why userAware?
- you shouldn't care about current user
- you can ask service for the real auth status for user id. it's important due to roles serialization